packages/components/pie-modal/src/index.ts

```diff
@@ -1,18 +1,40 @@
-import { LitElement, html, unsafeCSS } from 'lit'; // eslint-disable-line import/no-extraneous-dependencies
-import { property } from 'lit/decorators.js';
-import { RtlMixin } from '@justeattakeaway/pie-webc-core';
+import { LitElement, unsafeCSS } from 'lit'; // eslint-disable-line import/no-extraneous-dependencies
+import { html, unsafeStatic } from 'lit/static-html.js';
+import { property } from 'lit/decorators.js'; // eslint-disable-line import/no-extraneous-dependencies
+import { RtlMixin, validPropertyValues, requiredProperty } from '@justeattakeaway/pie-webc-core';
 
 import styles from './modal.scss?inline';
+import { HEADING_LEVELS } from './defs';
+
+// Valid values available to consumers
+export { HEADING_LEVELS };
+
+const componentSelector = 'pie-modal';
 
 export class PieModal extends RtlMixin(LitElement) {
     @property({ type: Boolean })
         isOpen = false;
 
-    // eslint-disable-next-line class-methods-use-this
+    @property({ type: String })
+    @requiredProperty(componentSelector)
+        heading!: string;
+
+    @property()
+    @validPropertyValues(componentSelector, Object.values(HEADING_LEVELS), HEADING_LEVELS.H2)
```

Comment:
I'm guessing we have some generic tests around the `validPropertyValues` decorator, but is it worth adding specific tests for modal where we pass in the heading types and explicitly test these usecases?

Just to cover if we ever hit an edge case that isn't covered by the generic function tests.